pkgs/development/libraries/gamin/default.nix

```diff
@@ -16,7 +16,7 @@ stdenv.mkDerivation (rec {
   # <sys/socket.h> with Glibc 2.9.
   configureFlags = "--disable-debug --with-python=${python} CPPFLAGS=-D_GNU_SOURCE";
 
-  patches = [ ./deadlock.patch ] ++ map fetchurl (import ./debian-patches.nix);
+  patches = [ ./deadlock.patch ./returnval.patch ] ++ map fetchurl (import ./debian-patches.nix);
```

Comment:
It's not a darwin specific patch, but it's pretty academic. Linux users will never miss it. Due to the way this function is used throughout the gamin code, it should be equivalent with and without the patch. So I've made it conditional on clang. But I'll happily undo and open against staging if you prefer.